From c9768e45e85667f228a6b376b6a664ac0dfaecfe Mon Sep 17 00:00:00 2001
From: Lori Burns <lori.burns@gmail.com>
Date: Tue, 18 Jul 2023 10:24:09 -0400
Subject: [PATCH 1/2] add netlib feature to builder

---
 build_libtensor.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build_libtensor.py b/build_libtensor.py
index 7ce16e58..c9b3fdbf 100755
--- a/build_libtensor.py
+++ b/build_libtensor.py
@@ -58,6 +58,8 @@ def configure(build_dir, source_dir, install_dir, build_type=None, features=[],
     if "mkl" in features:
         # Use sequential Intel MKL version
         args += ["-DBLA_VENDOR=Intel10_64lp_seq"]
+    if "netlib" in features:
+        args += ["-DBLA_VENDOR=Generic"]
 
     subprocess.check_call(args + [source_dir], cwd=build_dir)
 
@@ -127,7 +129,7 @@ def main():
                         help="The build type to configure.")
     parser.add_argument("--features", default=[],
                         nargs="+", help="Select optional features for build.",
-                        choices=["libxm", "mkl", "mpi"])
+                        choices=["libxm", "mkl", "mpi", "netlib"])
     parser.add_argument("--documentation", default=False, action="store_true",
                         help="Build documentation using doxygen.")
 
-- 
2.39.2 (Apple Git-143)

